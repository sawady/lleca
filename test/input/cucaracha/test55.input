
fun suma_vectores(v : Vec, w : Vec) {
    i := 0
    while i < #v {
        v[i] := v[i] + w[i]
        i := i + 1
    }
}

fun printVector(v : Vec) {
    i := 0
    putChar(91)
    while i < #v {
        putNum(v[i])
        i := i + 1
        if i < #v {
            putChar(44)
            putChar(32)
        }
    }
    putChar(93)
    putChar(10)
}

fun main() {
    v := [1, 2, 3, 4, 5, 6, 7, 8, 9]
    w := [10, 20, 30, 40, 50, 60, 70, 80, 90]
    printVector(v)
    printVector(w)
    suma_vectores(v, w)
    printVector(v)
    printVector(w)
    suma_vectores(v, v)
    printVector(v)
    printVector(w)
    suma_vectores(w, w)
    printVector(v)
    printVector(w)
}
