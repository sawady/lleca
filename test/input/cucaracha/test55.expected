Cons(
  Function(
    suma_vectores,
    Cons(
      Parameter(
        v,
        Vec
      ),
      Cons(
        Parameter(
          w,
          Vec
        ),
        Nil
      )
    ),
    Unit,
    Cons(
      StmtAssign(
        i,
        ExprConstNum(
          0
        )
      ),
      Cons(
        StmtWhile(
          ExprLt(
            ExprVar(
              i
            ),
            ExprVecLength(
              v
            )
          ),
          Cons(
            StmtVecAssign(
              v,
              ExprVar(
                i
              ),
              ExprAdd(
                ExprVecDeref(
                  v,
                  ExprVar(
                    i
                  )
                ),
                ExprVecDeref(
                  w,
                  ExprVar(
                    i
                  )
                )
              )
            ),
            Cons(
              StmtAssign(
                i,
                ExprAdd(
                  ExprVar(
                    i
                  ),
                  ExprConstNum(
                    1
                  )
                )
              ),
              Nil
            )
          )
        ),
        Nil
      )
    )
  ),
  Cons(
    Function(
      printVector,
      Cons(
        Parameter(
          v,
          Vec
        ),
        Nil
      ),
      Unit,
      Cons(
        StmtAssign(
          i,
          ExprConstNum(
            0
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                91
              ),
              Nil
            )
          ),
          Cons(
            StmtWhile(
              ExprLt(
                ExprVar(
                  i
                ),
                ExprVecLength(
                  v
                )
              ),
              Cons(
                StmtCall(
                  putNum,
                  Cons(
                    ExprVecDeref(
                      v,
                      ExprVar(
                        i
                      )
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtAssign(
                    i,
                    ExprAdd(
                      ExprVar(
                        i
                      ),
                      ExprConstNum(
                        1
                      )
                    )
                  ),
                  Cons(
                    StmtIf(
                      ExprLt(
                        ExprVar(
                          i
                        ),
                        ExprVecLength(
                          v
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprConstNum(
                              44
                            ),
                            Nil
                          )
                        ),
                        Cons(
                          StmtCall(
                            putChar,
                            Cons(
                              ExprConstNum(
                                32
                              ),
                              Nil
                            )
                          ),
                          Nil
                        )
                      ),
                      Nothing
                    ),
                    Nil
                  )
                )
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    93
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                ),
                Nil
              )
            )
          )
        )
      )
    ),
    Cons(
      Function(
        main,
        Nil,
        Unit,
        Cons(
          StmtAssign(
            v,
            ExprVecMake(
              Cons(
                ExprConstNum(
                  1
                ),
                Cons(
                  ExprConstNum(
                    2
                  ),
                  Cons(
                    ExprConstNum(
                      3
                    ),
                    Cons(
                      ExprConstNum(
                        4
                      ),
                      Cons(
                        ExprConstNum(
                          5
                        ),
                        Cons(
                          ExprConstNum(
                            6
                          ),
                          Cons(
                            ExprConstNum(
                              7
                            ),
                            Cons(
                              ExprConstNum(
                                8
                              ),
                              Cons(
                                ExprConstNum(
                                  9
                                ),
                                Nil
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            )
          ),
          Cons(
            StmtAssign(
              w,
              ExprVecMake(
                Cons(
                  ExprConstNum(
                    10
                  ),
                  Cons(
                    ExprConstNum(
                      20
                    ),
                    Cons(
                      ExprConstNum(
                        30
                      ),
                      Cons(
                        ExprConstNum(
                          40
                        ),
                        Cons(
                          ExprConstNum(
                            50
                          ),
                          Cons(
                            ExprConstNum(
                              60
                            ),
                            Cons(
                              ExprConstNum(
                                70
                              ),
                              Cons(
                                ExprConstNum(
                                  80
                                ),
                                Cons(
                                  ExprConstNum(
                                    90
                                  ),
                                  Nil
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            ),
            Cons(
              StmtCall(
                printVector,
                Cons(
                  ExprVar(
                    v
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  printVector,
                  Cons(
                    ExprVar(
                      w
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    suma_vectores,
                    Cons(
                      ExprVar(
                        v
                      ),
                      Cons(
                        ExprVar(
                          w
                        ),
                        Nil
                      )
                    )
                  ),
                  Cons(
                    StmtCall(
                      printVector,
                      Cons(
                        ExprVar(
                          v
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtCall(
                        printVector,
                        Cons(
                          ExprVar(
                            w
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          suma_vectores,
                          Cons(
                            ExprVar(
                              v
                            ),
                            Cons(
                              ExprVar(
                                v
                              ),
                              Nil
                            )
                          )
                        ),
                        Cons(
                          StmtCall(
                            printVector,
                            Cons(
                              ExprVar(
                                v
                              ),
                              Nil
                            )
                          ),
                          Cons(
                            StmtCall(
                              printVector,
                              Cons(
                                ExprVar(
                                  w
                                ),
                                Nil
                              )
                            ),
                            Cons(
                              StmtCall(
                                suma_vectores,
                                Cons(
                                  ExprVar(
                                    w
                                  ),
                                  Cons(
                                    ExprVar(
                                      w
                                    ),
                                    Nil
                                  )
                                )
                              ),
                              Cons(
                                StmtCall(
                                  printVector,
                                  Cons(
                                    ExprVar(
                                      v
                                    ),
                                    Nil
                                  )
                                ),
                                Cons(
                                  StmtCall(
                                    printVector,
                                    Cons(
                                      ExprVar(
                                        w
                                      ),
                                      Nil
                                    )
                                  ),
                                  Nil
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      ),
      Nil
    )
  )
)
